delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/http/HttpClient.scala

```diff
@@ -33,83 +30,95 @@ trait HttpClient {
+  def fromJsonTo[A: Decoder: ClassTag](req: HttpRequest): IO[A] =
     fromEntityTo(req)(decoderUnmarshaller[A], implicitly[ClassTag[A]])
 
   /**
     * Execute the argument request and unmarshal the response into an A using an [[Unmarshaller]].
     */
-  def fromEntityTo[A: FromEntityUnmarshaller: ClassTag](req: HttpRequest): HttpResult[A]
+  def fromEntityTo[A: FromEntityUnmarshaller: ClassTag](req: HttpRequest): IO[A]
 
   /**
     * Execute the argument request and return the stream of [[ByteString]].
     */
-  def toDataBytes(req: HttpRequest): HttpResult[AkkaSource]
+  def toDataBytes(req: HttpRequest): IO[AkkaSource]
 
   /**
     * Execute the argument request, consume the response and ignore, returning the passed ''returnValue'' when the
     * response is successful.
     */
-  def discardBytes[A](req: HttpRequest, returnValue: => A): HttpResult[A]
+  def discardBytes[A](req: HttpRequest, returnValue: => A): IO[A]
 
 }
 
 object HttpClient {
 
-  type HttpResult[A] = BIO[HttpClientError, A]
-
   private val acceptEncoding =
     AcceptEncoding.create(HttpEncodingRange.create(HttpEncodings.gzip), HttpEncodingRange.create(HttpEncodings.deflate))
 
   private[http] trait HttpSingleRequest {
-    def execute(request: HttpRequest): Task[HttpResponse]
+    def execute(request: HttpRequest): IO[HttpResponse]
   }
 
   private[http] object HttpSingleRequest {
-    def default(implicit as: ActorSystem): HttpSingleRequest =
-      (request: HttpRequest) => Task.deferFuture(Http().singleRequest(request))
+    def default(implicit as: ActorSystem, cs: ContextShift[IO]): HttpSingleRequest =
+      (request: HttpRequest) => IO.fromFuture(IO.delay(Http().singleRequest(request)))
   }
 
   /**
     * Construct the Http client using an underlying akka http client
     */
-  final def apply()(implicit httpConfig: HttpClientConfig, as: ActorSystem, scheduler: Scheduler): HttpClient = {
+  final def apply()(implicit
+      httpConfig: HttpClientConfig,
+      as: ActorSystem,
+      ec: ExecutionContext,
+      timer: Timer[IO],
+      cs: ContextShift[IO]
+  ): HttpClient = {
     apply(HttpSingleRequest.default)
   }
 
   /**
     * Construct an Http client using an underlying akka http client which will not retry on failures
     */
-  final def noRetry(compression: Boolean)(implicit as: ActorSystem, scheduler: Scheduler): HttpClient = {
+  final def noRetry(
+      compression: Boolean
+  )(implicit as: ActorSystem, ec: ExecutionContext, timer: Timer[IO], cs: ContextShift[IO]): HttpClient = {
     implicit val config: HttpClientConfig = HttpClientConfig.noRetry(compression)
     apply()
   }
 
   private[http] def apply(
       client: HttpSingleRequest
-  )(implicit httpConfig: HttpClientConfig, as: ActorSystem, scheduler: Scheduler): HttpClient =
+  )(implicit
+      httpConfig: HttpClientConfig,
+      as: ActorSystem,
+      ec: ExecutionContext,
+      timer: Timer[IO],
+      cs: ContextShift[IO]
```

Comment:
Retry

Reply:
Could be done in a later PR but I think we don't use the retry strategy anymore in the HttpClient (we always pass the `alwaysGiveUp`)
The reason behind that is it was creating more trouble than it solved because this one is very low-level and could be included in other retries, creating in the end timeouts for clients, etc...